ftplugin/supercollider.vim

```diff
@@ -194,33 +204,52 @@ function SClang_line()
 endfunction
 
 function SClang_block()
-	let [blkstart,blkend] = FindOuterMostBlock()
-	"blkstart[0],blkend[0] call SClang_send()
-	"these next lines are just a hack, how can i do the above??
-	let cmd = blkstart[0] . "," . blkend[0] . " call SClang_send()"
-	let l:origline = line(".")
-	let l:origcol = col(".")
-	exe cmd
-	call cursor(l:origline,l:origcol)
-	call FlashRegion(blkstart[0], blkend[0])
+  let [blkstart,blkend] = FindOuterMostBlock()
+  "blkstart[0],blkend[0] call SClang_send()
+  "these next lines are just a hack, how can i do the above??
+  let cmd = blkstart[0] . "," . blkend[0] . " call SClang_send()"
+  let l:origline = line(".")
+  let l:origcol = col(".")
+  exe cmd
+  call cursor(l:origline,l:origcol)
+  call FlashRegion(blkstart[0], blkend[0])
 endfunction
 
 " ========================================================================================
 
 let s:sclangStarted = 0
 
-function SClangStart()
-    if $TERM[0:5] == "screen"
-        if executable("tmux")
-            call system("tmux split-window -p 20 ; tmux send-keys " . s:sclangPipeApp . " Enter ; tmux select-pane -U")
-            let s:sclangStarted = 1
-        else
-            echo "Sorry, screen is not supported yet.."
-        endif
-    else
-        call system(s:sclangTerm . " " . s:sclangPipeApp . "&")
-        let s:sclangStarted = 1
+function SClangStart(...)
+  let l:tmux = exists('$TMUX')
+  let l:screen = exists('$STY')
+  if l:tmux || l:screen
+    let l:splitDir = (a:0 == 2) ? a:1 : s:scSplitDirection
+    let l:splitSize = (a:0 == 2) ? a:2 : s:scSplitSize
+
+    if l:tmux
+      let l:cmd = "tmux split-window -" . l:splitDir . " -p " . l:splitSize . " ;"
+      let l:cmd .= "tmux send-keys " . s:sclangPipeApp . " Enter ; tmux select-pane -l"
+      call system(l:cmd)
+    endif
+
+    if l:screen
```

Comment:
Might get a little crazy if one runs tmux in screen or screen in tmux, but we can fix this, once we get there...